client/battle/BattleFieldController.cpp

```diff
@@ -243,7 +243,7 @@ std::set<BattleHex> BattleFieldController::getMovementRangeForHoveredStack()
 	if (!owner.stacksController->getActiveStack())
 		return result;
 
-	if (!settings["battle"]["movementHighlightOnHover"].Bool())
+	if (!settings["battle"]["movementHighlightOnHover"].Bool() && !GH.isKeyboardShiftDown())

```

Comment:
how to deal with touchscreen?

Reply:
This check existed before, I only moved it in here from keyPressed. 

And don't think that we can display movement highlight on *cursor hovering* on touchscreen anyway. So desktop-only feature both before my changes and after.